core/src/main/java/com/google/zxing/aztec/decoder/Decoder.java

```diff
@@ -332,6 +333,22 @@ private static int readCode(boolean[] rawbits, int startIndex, int length) {
     return res;
   }
 
+  private static byte boolsToByte(final boolean[] array, final int start) {
+    byte b = 0;
+    for (int i = 0; i < 8; i++) {
+      if (array[start + i])
+        b |= 1 << (7 - i);
+    }
+    return b;
+  }
+
+  public static byte[] convertBoolArrayToByteArray(boolean[] boolArr) {
+    byte[] byteArr = new byte[boolArr.length / 8];
```

Comment:
I certainly can add the missing braces and resubmit, but I don't feel comfortable changing the actual code, as it's a bit over my head. Maybe @iSDP can weigh in on this or add the checks you're talking about.